docs/getting-started.md

```diff
@@ -0,0 +1,340 @@
+# Getting started
+
+For existing code examples, please check out our [example project](/examples).
+
+## Installation
+
+Installation via [JitPack](https://jitpack.io/) and [Gradle](https://gradle.org/).
+
+#### 1. Enable JitPack repository in your build configuration
+```kotlin
+// JitPack repository
+repositories {
+    maven { url = uri("https://jitpack.io") }
+}
+```
+
+#### 2. Add KSMT core dependency
+```kotlin
+dependencies {
+    // core 
+    implementation("com.github.UnitTestBot.ksmt:ksmt-core:0.2.0")    
+}
+```
+
+#### 3. Add one or more SMT solver dependencies
+```kotlin
+dependencies {
+    // z3 
+    implementation("com.github.UnitTestBot.ksmt:ksmt-z3:0.2.0")
+    // bitwuzla
+    implementation("com.github.UnitTestBot.ksmt:ksmt-bitwuzla:0.2.0")
+}
+```
+SMT solver specific packages are provided with solver native binaries. 
+
+## KSMT usage
+
+First, create a KSMT context that manages all expressions and solvers.
+```kotlin
+val ctx = KContext()
+```
+
+### Working with SMT formulas
+
+Once context is initialized, we can create an expressions. 
+In this example, we want to create an expression 
+
+`a && (b >= c + 3)`
+
+over boolean variable `a` and integer variables `b` and `c`.
+
+```kotlin
+with(ctx) {
+    // create symbolic variables
+    val a by boolSort
+    val b by intSort
+    val c by intSort
+
+    // create expression
+    val constraint = a and (b ge c + 3.intExpr)
+}
+```
+All KSMT expressions are typed and incorrect terms (e.g. `and` with integer arguments) 
+will result in a compile-time error.
+
+### Working with SMT solvers
+
+To check SMT formula satisfiability we need to instantiate an SMT solver. 
+In this example, we use `constraint` from a previous step as an SMT formula 
+and we use Z3 as an SMT solver.
+
+```kotlin
+with(ctx) {
+    KZ3Solver(this).use { solver -> // create s Z3 Smt solver instance
+        // assert expression
+        solver.assert(constraint)
+        
+        // check assertions satisfiability with timeout
+        val satisfiability = solver.check(timeout = 1.seconds)
+        println(satisfiability) // SAT
+        
+        // obtain model
+        val model = solver.model()
+        
+        println("$a = ${model.eval(a)}") // a = true
+        println("$b = ${model.eval(b)}") // b = 0
+        println("$c = ${model.eval(c)}") // c = -3
+    }
+}
+```
+The formula in the example above is satisfiable and we can obtain a model. 
+The model contains a concrete values of our symbolic variables `a`, `b` and `c`
+which evaluate our formula to `true`.
+
+Note the use of kotlin `.use { }` construction 
+which is useful for releasing all native resources acquired by the solver.
+
+### Solver incremental api
+
+KSMT solver api provides two approaches to incremental formula solving: push/pop
```

Comment:
KSMT solver API provides two approaches to incremental formula solving: push/pop